numbat/src/parser.rs

```diff
@@ -1152,7 +1152,28 @@ impl<'a> Parser<'a> {
             let mut parts = vec![StringPart::Fixed(strip_first_and_last(&token.lexeme))];
 
             let expr = self.expression()?;
-            parts.push(StringPart::Interpolation(expr.full_span(), Box::new(expr)));
+
+            let precision = if let Some(precision_token) =
+                self.match_exact(TokenKind::StringInterpolationSpecifier)
+            {
+                Some(
+                    precision_token
+                        .lexeme
+                        .chars()
+                        .skip(3)
+                        .collect::<String>()
+                        .parse::<u64>()
+                        // TODO
+                        .unwrap(),
+                )
+            } else {
+                None
+            };
+            parts.push(StringPart::Interpolation {
+                span: expr.full_span(),
+                expr: Box::new(expr),
+                precision,
+            });
```

Comment:
This code is very obviously horrible (also only handles `:0.3` and not `:.3`), but I need to think some more about how to improve it.